### PR TITLE
[FOGL-1928] delete south service & north instance api support

### PR DESCRIPTION
**Pull request by praveen-garg**
Test scenarios:

1) enabled service
2) disabled service
3) a service with shutdown status [add, enable and (disable or shutdown via curl)]

4) enabled north task (instance)
5) disabled north task (instance)

### CODE REVIEW

File: python/foglamp/services/core/service_registry/service_registry.py [hunk 1]
```diff
@@ -103,6 +103,7 @@ def unregister(cls, service_id):
         """
         expunged_service = cls._expunge(service_id, ServiceRecord.Status.Down)
         cls._logger.info("Stopped {}".format(str(expunged_service)))
+        cls.remove_from_registry(service_id)
```
Comment (praveen-garg) [inline python/foglamp/services/core/service_registry/service_registry.py]: May be, We should move this as first action to delete service method, if service's status is DOWN then remove from service registry.

File: python/foglamp/services/core/api/service.py [hunk 2]
```diff
@@ -69,6 +69,50 @@ def get_service_records():
...
+        import uuid
+        sch_id = uuid.UUID(svc_schedule['id'])
+        if svc_schedule['enabled'].lower() == 't':
+            # disable it
+            await server.Server.scheduler.disable_schedule(sch_id)
+        # delete it
+        await server.Server.scheduler.delete_schedule(sch_id)
+
+        # delete config for the service name
+        await revert_configuration(storage, svc)  # Delete configuration entry
```
Comment (MarkRiddoch) [inline python/foglamp/services/core/api/service.py]: We should also remove any child categories of the category whose name matches the service.

**Comment by praveen-garg** [comment]
@MarkRiddoch  feedback changes done.  Also fixed for add service, as it was missing `for <name>Advanced`

**Comment by praveen-garg** [comment]
#1201 is provided on top of this for north task instance deletion;

File: python/foglamp/services/core/api/service.py [hunk 3]
```diff
@@ -236,14 +280,34 @@ def get_service_records():
...
-async def revert_configuration(storage, key):
+async def get_schedule(storage, schedule_name):
+    payload = PayloadBuilder().SELECT(["id", "enabled"]).WHERE(['schedule_name', '=', schedule_name]).payload()
+    result = await storage.query_tbl_with_payload('schedules', payload)
+    return result
+
+
+async def delete_configuration(storage, key):
+    await delete_configuration_category(storage, key)
+    await delete_configuration_category(storage, "{}Advanced".format(key))
```
Comment (MarkRiddoch) [inline python/foglamp/services/core/api/service.py]: It should not just be a fixed name. What I am saying is that you need to find all the children of this category and delete them all. So go to the category_children table, get all children of this category and delete all of those categories. For each of those categories repeat this until you find no new children. Then remove the entries from the category_children table.

This code should not have any hardwired name, like Advanced in it all all.

Comment (praveen-garg) [reply python/foglamp/services/core/api/service.py]: Well, {}Advanced is being deleted from `configuration` table itself. which is required as we have that hardwired insertion via south->ingest.py `        category = "{}Advanced".format(cls._parent_service._name)`

> get all children of this category and delete all of those categories

next 2 def calls are doing the same: delete_parent_child_configuration, delete_advance_child_configuration

I did not find any config thing (including) left in `configuration` or `category_children` table after these calls.

Comment (praveen-garg) [reply python/foglamp/services/core/api/service.py]: @MarkRiddoch  FOGL-2014 created.

**Pull request merged**